File: src/programmersguide/workingwithcharts/charttrendlines/java/ChartTrendLines.java
/* 
 * Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
 *
 * This file is part of Aspose.Slides. The source code in this file
 * is only intended as a supplement to the documentation, and is provided
 * "as is", without warranty of any kind, either expressed or implied.
 */
 
package programmersguide.workingwithcharts.charttrendlines.java;

import com.aspose.slides.*;

import java.awt.*;

public class ChartTrendLines
{
    public static void main(String[] args) throws Exception
    {
        // The path to the documents directory.
        String dataDir = "src/programmersguide/workingwithcharts/charttrendlines/data/";

        //Creating empty presentation//Creating empty presentation
        Presentation pres = new Presentation();

//Creating a clustered column chart
        IChart chart = pres.getSlides().get_Item(0).getShapes().addChart(ChartType.ClusteredColumn, 20, 20, 500, 400);

//Adding ponential trend line for chart series 1
        ITrendline tredLinep = chart.getChartData().getSeries().get_Item(0).getTrendLines().add(TrendlineType.Exponential);
        tredLinep.setDisplayEquation(false);
        tredLinep.setDisplayRSquaredValue(false);

//Adding Linear trend line for chart series 1
        ITrendline tredLineLin = chart.getChartData().getSeries().get_Item(0).getTrendLines().add(TrendlineType.Linear);
        tredLineLin.setTrendlineType(TrendlineType.Linear);
        tredLineLin.getFormat().getLine().getFillFormat().setFillType(FillType.Solid);
        tredLineLin.getFormat().getLine().getFillFormat().getSolidFillColor().setColor(Color.RED);


//Adding Logarithmic trend line for chart series 2
        ITrendline tredLineLog = chart.getChartData().getSeries().get_Item(1).getTrendLines().add(TrendlineType.Logarithmic);
        tredLineLog.setTrendlineType(TrendlineType.Logarithmic);
        tredLineLog.addTextFrameForOverriding("New log trend line");

//Adding MovingAverage trend line for chart series 2
        ITrendline tredLineMovAvg = chart.getChartData().getSeries().get_Item(1).getTrendLines().add(TrendlineType.MovingAverage);
        tredLineMovAvg.setTrendlineType(TrendlineType.MovingAverage);
        tredLineMovAvg.setPeriod((byte)3);
        tredLineMovAvg.setTrendlineName("New TrendLine Name");

//Adding Polynomial trend line for chart series 3
        ITrendline tredLinePol = chart.getChartData().getSeries().get_Item(2).getTrendLines().add(TrendlineType.Polynomial);
        tredLinePol.setTrendlineType(TrendlineType.Polynomial);
        tredLinePol.setForward(1);
        tredLinePol.setOrder ((byte)3);

//Adding Power trend line for chart series 3
        ITrendline tredLinePower = chart.getChartData().getSeries().get_Item(1).getTrendLines().add(TrendlineType.Power);
        tredLinePower.setTrendlineType(TrendlineType.Power);
        tredLinePower.setBackward(1);

//Saving presentation
        pres.save(dataDir+ "ChartTrendLines.pptx", SaveFormat.Pptx);
    }
}

File: Plugins/Aspose_Slides_Java_for_Ruby/lib/asposeslidesjava/Charts/charttrendlines.rb
module Asposeslidesjava
  module ChartTrendLines
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Creating empty presentation
        pres = Rjb::import('com.aspose.slides.Presentation').new

        # Creating a clustered column chart
        chart = pres.getSlides().get_Item(0).getShapes().addChart(Rjb::import('com.aspose.slides.ChartType').ClusteredColumn, 20, 20, 500, 400)

        # Adding ponential trend line for chart series 1
        tredLinep = chart.getChartData().getSeries().get_Item(0).getTrendLines().add(Rjb::import('com.aspose.slides.TrendlineType').Exponential)
        tredLinep.setDisplayEquation(false)
        tredLinep.setDisplayRSquaredValue(false)

        # Adding Linear trend line for chart series 1
        tredLineLin = chart.getChartData().getSeries().get_Item(0).getTrendLines().add(Rjb::import('com.aspose.slides.TrendlineType').Linear)
        tredLineLin.setTrendlineType(Rjb::import('com.aspose.slides.TrendlineType').Linear)
        tredLineLin.getFormat().getLine().getFillFormat().setFillType(Rjb::import('com.aspose.slides.FillType').Solid)
        tredLineLin.getFormat().getLine().getFillFormat().getSolidFillColor().setColor(Rjb::import('java.awt.Color').RED)


        # Adding Logarithmic trend line for chart series 2
        tredLineLog = chart.getChartData().getSeries().get_Item(1).getTrendLines().add(Rjb::import('com.aspose.slides.TrendlineType').Logarithmic)
        tredLineLog.setTrendlineType(Rjb::import('com.aspose.slides.TrendlineType').Logarithmic)
        tredLineLog.addTextFrameForOverriding("New log trend line")

        # Adding MovingAverage trend line for chart series 2
        tredLineMovAvg = chart.getChartData().getSeries().get_Item(1).getTrendLines().add(Rjb::import('com.aspose.slides.TrendlineType').MovingAverage)
        tredLineMovAvg.setTrendlineType(Rjb::import('com.aspose.slides.TrendlineType').MovingAverage)
        tredLineMovAvg.setPeriod(3)
        tredLineMovAvg.setTrendlineName("New TrendLine Name")

        # Adding Polynomial trend line for chart series 3
        tredLinePol = chart.getChartData().getSeries().get_Item(2).getTrendLines().add(Rjb::import('com.aspose.slides.TrendlineType').Polynomial)
        tredLinePol.setTrendlineType(Rjb::import('com.aspose.slides.TrendlineType').Polynomial)
        tredLinePol.setForward(1)
        tredLinePol.setOrder(3)

        # Adding Power trend line for chart series 3
        tredLinePower = chart.getChartData().getSeries().get_Item(1).getTrendLines().add(Rjb::import('com.aspose.slides.TrendlineType').Power)
        tredLinePower.setTrendlineType(Rjb::import('com.aspose.slides.TrendlineType').Power)
        tredLinePower.setBackward(1)

        # Saving the presentation
        save_format = Rjb::import('com.aspose.slides.SaveFormat')
        pres.save(data_dir + "ChartTrendLines.pptx", save_format.Pptx)

        puts "Done with chart trend lines, please check the output file."
    end    
  end
end
